2015-Fall/lrusnac/CalculatorV2/CalculatorV2/CalculatorViewController.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  CalculatorViewController.swift
+//  CalculatorV2
+//
+//  Created by Leonid Rusnac on 08/09/15.
+//  Copyright (c) 2015 Leonid Rusnac. All rights reserved.
+//
+
+import UIKit
+
+class CalculatorViewController: UIViewController {
+
+    @IBOutlet weak var display: UILabel!
+    
+    var calculatorModel = CalculatorModel()
+    var userIsInTheMiddleOfTypingANumber = false
+    var pointWasInsertedInThisNumber = false
+    
+    @IBAction func appendDigit(sender: UIButton) {
+        if let digit = sender.currentTitle {
+            if userIsInTheMiddleOfTypingANumber {
+                display.text = display.text! + digit
+            } else {
+                display.text = digit
+                userIsInTheMiddleOfTypingANumber = true
+            }
+        }
+    }
+
+    @IBAction func clear() {
+        calculatorModel.clear()
+        display.text = "\(0)"
+        userIsInTheMiddleOfTypingANumber = false
+        pointWasInsertedInThisNumber = false
+    }
+    
+    @IBAction func backspace() {
+        // I let the user to backspace only if it is in middle of typing
+        if userIsInTheMiddleOfTypingANumber {
+            if let displayText = display.text {
+                let newDisplay = dropLast(displayText)
+                if newDisplay.isEmpty {
+                    userIsInTheMiddleOfTypingANumber = false
+                    pointWasInsertedInThisNumber = false
+                    display.text = "0"
+                } else {
+                    if let pointIndex = newDisplay.rangeOfString(".") {
```

Comment:
sure it can but this mean search inside the string every time instead of looking to a bool.
therefor there is the cost of updating the boolean.

I'm not sure what is the best practice in swift, but I think they are similar solutions